src/Calendar.php

```diff
@@ -97,17 +97,25 @@ public function __construct(
         $this->status = new Status($this->deriveStatusTypeFromSubEvents(), []);
     }
 
-    public function withStatus(Status $status): self
+    /**
+     * Set $updateStatusInTimestamps to true when setting a new top status that should also be applied to all subEvents.
+     * Set it to false when constructing a Calendar object that's deserialized from JSON and you want to set the raw
+     * status without affecting the status in the subEvents.
+     */
+    public function withStatus(Status $status, bool $updateStatusInTimestamps = true): self
```

Comment:
@jdrieghe I discussed potential fixes for this problem of `withStatus()` always overwriting the status in the subEvents with @LucWollants and we came up with several solutions:

- Rename `withStatus()` to something more explicit that indicates that it will overwrite the status in the subEvents and add another `withStatus()` that only overwrites the top status (to use when creating a Calendar object from incoming JSON)
- Adding an optional `Status` argument to the constructor to use when to use when creating a Calendar object from incoming JSON, which will again only set the top status. If none is supplied we derive the status as we already do.
- Adding this boolean toggle, which I did for now since it was the least effort so we have something working and can then refactor it to a better solution after discussing with you.

So our question is: What do you think would be a good, explicit, solution to make it possible to either set a status on the top level, or also make it trickle down to the subEvents in some cases? My preference would go to having two separate methods, but we're unsure so far about the best naming and this would also involve making changes to udb3-silex to change the current usages of `withStatus`.

Reply:
@LucWollants also suggested `applyStatus()` which would overwrite the status in the subEvents as well (i.e. an explicit update) and `withStatus()` which would just set the top status property. It's a good idea in the right direction I think but doesn't feel explicit enough yet when looking at a glance imho.

Reply:
I don't think this approach is "that bad" because the optional `$updateStatusInTimestamps` explains what will happen.

Reply:
My issue with the boolean is that it's not that clear from the outside. For example in the test (and later on where we'll use it in the `CalendarJSONDeserializer`) you just see a `false` without knowing what it does at a glance. It's also optional now for backwards compatibility with the existing usages in `udb3-silex`, so it's easy to accidentally make the mistake we had in `deserialize()`.

Reply:
Another thought I just had: We could make `withStatus()` only update the top status, and extract the logic for updating the status in the subEvents to an extra `withStatusOnSubEvents()` that only does that? So each method has a single responsibility. When parsing the full calendar object from JSON (major-info endpoint and json imports for example) we'd only use `withStatus()` then to set the top status. And in the `UpdateStatus` command handler we'd use `withStatus()` + `withStatusOnSubEvents`?

Reply:
I like that approach, having 2 dedicated methods `withStatus` and `withStatusOnSubEvents`. I would keep `withStatusOnSubEvents` inside the `Calendar`.